onnx/defs/schema.h

```diff
@@ -659,14 +704,27 @@ class OpSchemaRegistry final {
   }
 };
 
-#define ONNX_OPERATOR_SCHEMA(name) \
-  ONNX_OPERATOR_SCHEMA_UNIQ_HELPER(__COUNTER__, name)
-#define ONNX_OPERATOR_SCHEMA_UNIQ_HELPER(Counter, name) \
-  ONNX_OPERATOR_SCHEMA_UNIQ(Counter, name)
-#define ONNX_OPERATOR_SCHEMA_UNIQ(Counter, name)                 \
-  static ONNX_NAMESPACE::OpSchemaRegistry::OpSchemaRegisterOnce( \
-      op_schema_register_once##name##Counter) =                  \
-      OpSchema(#name, __FILE__, __LINE__)
+// Registers all schema of a given operator set
+template<class T> 
+void RegisterOpSetSchema(){
+  T::ForEachSchema([](OpSchema&& schema) {OpSchemaRegistry::OpSchemaRegisterOnce(registration) = std::forward<OpSchema&&>(schema); });
+};
+
+// Forward declaration for the non-specialized GetOpSchema method.  This enforces a consistent
+// signature on functions that query individual schema, which are defined as specializations
+// of this function.
+template <typename T> OpSchema GetOpSchema();
+
+// Defines specialization of GetOpSchema for a class whose name is determined based on 
+// a convention using name, domain, and version.
+#define ONNX_OPERATOR_SCHEMA(name, domain, ver, impl) \
```

Comment:
most schemas are of the onnx core domain. can you change this to a helper macro and leave `ONNX_OPERATOR_SCHEMA` dedicated to the onnx core domain? For ONNX_ML I think `ONNX_ML_OPERATOR_SCHEMA` can be used.

Reply:
I can do that